### ISSUE TITLE
Defaults don't work with STI

### ISSUE THREAD

**Issue by DRBragg** [palkan/store_attribute#43]
I'm assuming this is an STI related bug but it could be a red herring, either way we're having an issue with defaults

### Tell us about your environment

**Ruby Version:** 7.1.3.4

**Rails Version:** 3.3.4

**PostgreSQL Version:** 15.4

**Store Attribute Version:** Attempting to update to 1.2 from 0.9.3

### What did you do?

We have an STI setup that looks a little like this:
```ruby
class Pages::PageSection < ApplicationRecord
	store_attribute :content, :heading_level, :string, default: "2"
end

class Pages::ProductBannerPageSection < Pages::PageSection
	store_attribute :content, :media_type, :string, default: "image"
end
```

in 0.9.3 we had this behavior for new records:
```ruby
Pages::ProductBannerPageSection.new.heading_level
=> "2"
Pages::ProductBannerPageSection.new.media_type
=> "image"
```
but in 1.2 we're seeing this:
```ruby
Pages::ProductBannerPageSection.new.heading_level
=> "2"
Pages::ProductBannerPageSection.new.media_type
=> nil
```

### What did you expect to happen?

We expect the defaults to continue to be set as defined.

### What actually happened?

It seems that only the defaults defined at the parent level are working. For some reason defaults defined on the children are not working.

**Comment by palkan** [comment]
Thanks for reporting!

Will try to take a look this week.

**Comment by palkan** [comment]
Hey @DRBragg,

I've tried to reproduce this behaviour in our tests but failed. Could you please take a look at the PR: https://github.com/palkan/store_attribute/pull/44 What am I missing?

P.S. Congrats with the RubyConf keynoting 💪

**Comment by DRBragg** [comment]
> Hey @DRBragg,
> 
> I've tried to reproduce this behaviour in our tests but failed. Could you please take a look at the PR: #44 What am I missing?
> 
> P.S. Congrats with the RubyConf keynoting 💪

Well frick... Thanks for digging in. The spec looks right for what we had boiled it down to. I'll play with it some and try to get it closer to what we have in prod. Maybe `store_attribute` is a red herring. 🤔 I'll let you know if I can reproduce. Thanks!

P.S. Thanks!! See you there?

**Comment by palkan** [comment]
> P.S. Thanks!! See you there?

Plan to make a final decision at or after Rails World (see you there?)

**Comment by DRBragg** [comment]
> > P.S. Thanks!! See you there?
> 
> Plan to make a final decision at or after Rails World (see you there?)

See you there!

**Comment by DRBragg** [comment]
@palkan We were able to update the spec in #44 to get the proper failure! It seems there was slightly more to it that we had originally thought. The trick was a _second_ `jsonb` column. Try this:

```ruby
# frozen_string_literal: true

connection = ActiveRecord::Base.connection

connection.drop_table "pages", if_exists: true

connection.transaction do
  connection.create_table("pages") do |t|
    t.string :title
    t.jsonb :content
    t.jsonb :design
    t.string :type
  end
end

class Page < ActiveRecord::Base
  store_attribute :content, :media_placement, :string, default: "right"
end

class MediaBannerPage < Page
  store_attribute :design, :heading_level, :string, default: "2"
  store_attribute :content, :media_type, :string, default: "image"
end
```

**Comment by palkan** [comment]
Thanks for the reproduction!

Fixed and released in v1.3.0.
